plugin/backend/ipvariant.d

```diff
@@ -0,0 +1,906 @@
+        if(sut.valid) {
+		
+		    final switch(cast(NamespaceType) ns.kind) with (NamespaceType) {
+	             case Normal:
+                         inner.hdr = modules.hdr.namespace(ns.name);
+                         inner.hdr.suppressIndent(1);
+                         inner.impl = modules.impl.namespace(ns.name);
+                         break;
+                     case TestDoubleSingleton:
+                         break;
+                     case TestDouble:
+                         break;
+                 }
+	    }
+        
+        if (sut.valid && isReqOrPro)
+        {
+            foreach (a; ns.funcRange) {
+                generateFuncImpl(a, inner.impl);
+            }
+
+            foreach (a; ns.classRange) {
+                foreach (b; a.methodPublicRange) { 
+                    b.visit!((const CppMethod a) => generateCppMeth(a, inner, ns.fullyQualifiedName, sut),
+                            (const CppMethodOp a) => writeln(""),
+                            (const CppCtor a) => generateCtor(a, inner),
+                            (const CppDtor a) => generateDtor(a, inner));
+                }          
+            }
+
+            if (!cifaces.canFind(currns) && ns.namespaceRange.length == 0 &&
+                    (currns_spl[$-1] == "Requirer" ||  currns_spl[$-1] == "Provider")) {
+
+                generateClass(inner, currns_spl[0..$-1], currns_spl[$-1], sut);
+                cifaces ~= currns;
+                writeln(cifaces);
+            }
+        }
+  
+        foreach (a; ns.namespaceRange) { 
+            //writeln("namespace "~currns~"::"~a.name);
+            eachNs(a, params, inner, inner_impl_singleton, lookup, cifaces);
+        }
+    }
+
+   //gmockGlobal(r.classRange, modules.gmock, params);
+    // no singleton in global namespace thus null
+    string[] cifaces;
+    foreach (a; r.namespaceRange()) {
+        eachNs(a, params, modules, null, (USRType usr) => container.find!LocationTag(usr), cifaces);
+    }
+}
+
+import std.container.array;
+@trusted Array!string getDataItems(SUTEnv sut) {
+    Array!string rarr; 
+    foreach(ciface; sut.iface.interfaces) {
+        foreach(ditem; ciface.ditems) {
+                rarr ~= ditem.name;
+        }
+    }
+
+    return rarr;
+}
+
+
+@trusted void generateClass(Generator.Modules inner, string[] ns, string type, SUTEnv sut) {
+    import std.array;
+    import std.string : toLower; 
+    
+    string fqn_ns = ns.join("::"); 
+    string impl_nsname = ns[$-1];
+    
+    with(inner.impl) {        
+        with (class_(impl_nsname ~ "_Impl", "I_" ~ impl_nsname)) {
+            with(private_) {
+                foreach(ciface; sut.iface.interfaces) {
+                    stmt(E(fqn_ns ~ "::" ~ ciface.name ~ "T " ~ toLower(ciface.name)));
+                }
+
+                stmt(E("RandomGenerator* randomGenerator"));
+            }
+            with(public_) {
+
+                with (ctor_body(impl_nsname ~ "_Impl")) { //Generate constructor
```

Comment:
I really should change dsrcgen to follow the D coding standard /regrets....
I'm sorry for the uglification of ya code :/